Weather/Views/WeatherView.swift

```diff
@@ -31,40 +29,114 @@ class WeatherView: UIView {
         backgroundImage.translatesAutoresizingMaskIntoConstraints = false
         addSubview(backgroundImage)
-        presenter?.activateBackgroundConstraints(view: backgroundImage)
+        activateBackgroundConstraints(view: backgroundImage)
         
         cityLabel.text = "Minsk"
         cityLabel.font = UIFont.boldSystemFont(ofSize: 34)
         cityLabel.translatesAutoresizingMaskIntoConstraints = false
         cityLabel.textColor = UIColor.white
         addSubview(cityLabel)
-        presenter?.activateCityLabelConstraints(view: cityLabel)
+        activateCityLabelConstraints(view: cityLabel)
         
         forecastStatusLabel.text = "Partly cloudly"
         forecastStatusLabel.font = UIFont.boldSystemFont(ofSize: 17)
         forecastStatusLabel.translatesAutoresizingMaskIntoConstraints = false
         forecastStatusLabel.textColor = UIColor.white
         addSubview(forecastStatusLabel)
-        presenter?.activateForecastStatusLabelConstraints(view: forecastStatusLabel, anchorView: cityLabel)
+        activateForecastStatusLabelConstraints(view: forecastStatusLabel, anchorView: cityLabel)
         
         currentTemperatureLabel.text = "2"
         currentTemperatureLabel.font = UIFont.boldSystemFont(ofSize: 54)
         currentTemperatureLabel.translatesAutoresizingMaskIntoConstraints = false
         currentTemperatureLabel.textColor = UIColor.white
         addSubview(currentTemperatureLabel)
-        presenter?.activateCurrentTemperatureLabelConstraints(view: currentTemperatureLabel, anchorView: forecastStatusLabel)
+        activateCurrentTemperatureLabelConstraints(view: currentTemperatureLabel, anchorView: forecastStatusLabel)
         
         timedForecastLayout.scrollDirection = .horizontal
         timedForecastLayout.sectionInset = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
         timedForecastLayout.itemSize = CGSize(width: 60, height: 70)
-        let timedForecastFrame = CGRect(x: 0, y: 200, width: UIScreen.main.bounds.width, height: 80)
+        let timedForecastFrame = CGRect(x: 0, y: 200, width: UIScreen.main.bounds.width, height: 90)
         timedForecastView = UICollectionView(frame: timedForecastFrame, collectionViewLayout: timedForecastLayout)
         timedForecastView.backgroundColor = UIColor.clear
         addSubview(timedForecastView)
-        presenter?.activateTimedForecastViewConstraints(view: timedForecastView, anchorView: currentTemperatureLabel)        
+        activateTimedForecastViewConstraints(view: timedForecastView, anchorView: currentTemperatureLabel)
+        
+        weekForecastLayout.scrollDirection = .vertical
+        weekForecastLayout.sectionInset = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+        weekForecastLayout.itemSize = CGSize(width: UIScreen.main.bounds.width, height: 70)
+        let weekForecastFrame = CGRect(x: 0, y: 300, width: UIScreen.main.bounds.width, height: 380)
```

Comment:
There is no need to set view's frame as you are using autolayout. Just call empty initializer `()`

Reply:
Still no need for setting frames by hands.

Reply:
@declipz If you mean collectionViewLayout and cell's frames then yes, you should calculate them manually as i remember. CollectionViewLayout if kind of complicated object, you should read some docs about the way it works.

Reply:
@pkondrashkov I'm actually a little confused and maybe can't imagine it. Would you send me some examples of it? Because all of my attempts were failed.

Reply:
> @pkondrashkov I'm actually a little confused and maybe can't imagine it. Would you send me some examples of it? Because all of my attempts were failed.

Found my mistake, but still can't resize it properly.